### PR TITLE
fix: k8s probes initialDelaySeconds to 0

### PR DESCRIPTION
**Pull request by edubacco**
using startup probe ensure readiness and liveness probes are not checked until the application is ready to handle them. There is no need to delay them.

### CODE REVIEW

**Review by giuscri — DISMISSED** [review]
LGTM: it's annoying to have probes delayed. They slow down the helm releases

**Comment by massicer** [comment]
I'm not sure this corresponds to a patch version increment. Does it seem like a breaking change WDYT? @edubacco

**Review by gi0baro — COMMENTED** [review]
Considering the `startup` probe is relatively new and probably not used by ANY of our application, I don't exactly agree with this.

Quite a few of our applications take several seconds to starts, removing the delay from probes is not exactly the "solution" IMHO.
In case the point here is to lower down the release time, why don't we just lower the default delay eg to 5-10 secs?

**Comment by edubacco** [comment]
> Considering the startup probe is relatively new and probably not used by ANY of our application
Actually every application of the tribe sellers uses the startup probe. I thought that others app was also using it..

> Quite a few of our applications take several seconds to starts, removing the delay from probes is not exactly the "solution" IMHO.
The intent is not removing the delay, but on the contrary, use the correct delay to every scenario; that's why the startup probe exists: the concept is that the _others_ probes starts as long as the startup probe gets the first success answer.
So you do not need anymore to put fixed delay, as the probes start to work as soon as possible: the helm release will be fast if the application starts fast, slow if the application starts slow.

In any case, they are default values, every application can use values it prefers. I just think that those kind of default values suggests to use startup probes

**Comment by gi0baro** [comment]
@edubacco 

> Actually every application of the tribe sellers uses the startup probe. I thought that others app was also using it..

Not the same in all the other tribes.

> 
> The intent is not removing the delay, but on the contrary, use the correct delay to every scenario; that's why the startup probe exists: the concept is that the _others_ probes starts as long as the startup probe gets the first success answer. So you do not need anymore to put fixed delay, as the probes start to work as soon as possible: the helm release will be fast if the application starts fast, slow if the application starts slow.

I agree, I'm not stating your principle is correct, I'm considering the average usage across the org of those values. Also consider all those probes are disabled by default, so unless you explicitly activate the startup probe, under default behaviour you need startup delays.

> In any case, they are default values, every application can use values it prefers. I just think that those kind of default values suggests to use startup probes

Agree, but since you're proposing a potential breaking change, I agree with @massicer that this should at least produce a minor version of those charts.

So, to summarise: I'll approve this with a proper version change :)

**Comment by edubacco** [comment]
Fixed versions: update to next minor! :D

**Review by massicer — APPROVED** [review]

**Review by gi0baro — APPROVED** [review]
